### PR TITLE
add a command to load a debug dump into postgres

### PR DESCRIPTION
**Pull request by jrockway**
I hacked this together so the code is not good.  But it is useful.

### CODE REVIEW

File: src/cmd/load-debug-dump-into-postgres/main.go [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+	"go.uber.org/zap"
+)
+
+var (
+	source = flag.String("source", "", "debug dump archive to import")
+	v      = flag.Bool("v", false, "verbose logs")
+	dsn    = flag.String("dsn", "host=localhost port=5432 database=pachydermlogs user=postgres pool_max_conns=50", "postgres dsn")
+)
+
+var nerr, nrows, ndups, nlines, njson, nfiles, nbinary, nbytesin, nbytesout, ngoro, nconnacq atomic.Int64
```
Comment (robert-uhl) [inline src/cmd/load-debug-dump-into-postgres/main.go:35]: _Minor_ nit: maybe camelCase these?

File: src/cmd/load-debug-dump-into-postgres/main.go [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+			err = errors.New("dump errored")
+		}
+		done(err)
+	}()
+
+	if *v {
+		log.SetLevel(log.DebugLevel)
+	}
+
+	ctx, c := pctx.Interactive()
```
Comment (robert-uhl) [inline src/cmd/load-debug-dump-into-postgres/main.go:148]: Should you just `defer c()` here? Doesn’t make a ton of different in `main`.

I think conventionally `cancel` might be clearer.

File: src/cmd/load-debug-dump-into-postgres/main.go [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+		nconnacq.Add(-1)
+		if err != nil {
+			log.Error(ctx, "cannot acquire db conn", zap.Error(err))
+			nerr.Add(1)
+			break
+		}
+
+		wg.Add(1)
+		go func() {
+			addFile(ctx, &buf, h.Name, c) //nolint:errcheck
```
Comment (robert-uhl) [inline src/cmd/load-debug-dump-into-postgres/main.go:199]: You _could_ collect errors … not a huge deal in a quick hack.

File: src/internal/log/loggers.go [hunk 4]
```diff
@@ -253,6 +253,7 @@ func InitBatchLogger(logFile string) func(err error) {
 			if keepLog {
 				zap.S().Infof("logfile retained at %v", logFile)
 			}
+			time.Sleep(5 * time.Millisecond)
```
Comment (robert-uhl) [inline src/internal/log/loggers.go:256]: Did you mean to retain this?

**Review by robert-uhl — COMMENTED** [review]
It looks pretty good. I wonder if this sort of quick-and-dirty hack might be a good candidate for a `src/internal/cmd`? Just a thought really.

File: src/internal/log/loggers.go (hunk 4 above)
Comment (jrockway) [reply src/internal/log/loggers.go:256]: Yeah, it seems to suffer the same fate of being "already closed" like the comment below is talking about even in this branch.

File: src/cmd/load-debug-dump-into-postgres/main.go (hunk 2 above)
Comment (jrockway) [reply src/cmd/load-debug-dump-into-postgres/main.go:148]: defer is too late for the success case (we want to print the "final stats" before exiting)

Comment (robert-uhl) [reply src/cmd/load-debug-dump-into-postgres/main.go:148]: You could close another channel and wait for it before finally exiting, but … that seems crazy, esp. given that this is in `main`. I’m cool with it as-is.

File: src/internal/log/loggers.go (hunk 4 above)
Comment (robert-uhl) [reply src/internal/log/loggers.go:256]: Ah, there’s a bigger issue then, with `InitBatchLogger` needing to be held open until all logging has completed. That doesn’t need to be resolved here — I’ve opend CORE-2237 to track it.

File: src/cmd/load-debug-dump-into-postgres/main.go (hunk 1 above)
Comment (jrockway) [reply src/cmd/load-debug-dump-into-postgres/main.go]: Done.

File: src/cmd/load-debug-dump-into-postgres/main.go (hunk 3 above)
Comment (jrockway) [reply src/cmd/load-debug-dump-into-postgres/main.go]: They're in the log; this causes the log to be retained and its filename to be printed.

**Comment by jrockway** [comment]
All feedback resolved.

**Review by robert-uhl — APPROVED** [review]

**Pull request merged**